### DISCUSSION TITLE
Emit Filters functionality extended to Maps

### DISCUSSION THREAD

**Discussion by jharner-sp** [apache/superset#18481] [Ideas]
The Emitting Filters flag is a godsend but I was hoping to get further use from it, in my case I have a map that I'd like to be able to also use as a filter (which would be augmented by [#13966](https://github.com/apache/superset/issues/13966) )
![filters_toggle](https://user-images.githubusercontent.com/82689532/120050303-c63eb300-bfd1-11eb-808f-b21157ca7d5c.PNG)
![map](https://user-images.githubusercontent.com/82689532/120050305-c76fe000-bfd1-11eb-92ae-a3c35a98b0e4.PNG)

**Comment by junlincc** [comment]
please elaborate your use case, what are you exactly trying to achieve in Explore and Dashboard for Country map chart. @jharding-blw 

allow user to emit filtering event to other charts on dashboard by clicking on the map? or drilldown into the next level on map?

**Comment by jharner-sp** [comment]
Allow the user to filter the dashboard via clicking on a state in the map. We have a lot of data spread across the US and using states to filter the rest of the dashboard allows ease of use by different departments given the customer's region/state.

**Comment by Octochef** [comment]
Hi @junlincc yes cross filters are absolutely key. I come from alot of kibana use and boy it feels weird not to be able to click on a map state and emit a filter state=CA or similar.

 Likewise not to be able to "comb" a time based visualization (e.g. click, select an area and release to create a time filter)  Where could we discuss all these? (not just the map here). I think its a wider initiative to get the widget to their most interactive. 
 
 Thanks for considering!

**Comment by jaljac** [comment]
Coming from PowerBI or other reporting tools this is something that I miss. I just begin with superset and when trying to create a dashboard with a map I realize that trying to reduce the information based on a map selection was not possible. This is something that makes the dashboard less powerful for analysis than those that you can create with other tools.

**Comment by romaiy** [comment]
I think I have found a solution to this problem. It is necessary to add 3 new behaviors in the index file country map in the metadata object

**Reply by romaiy** [reply]
In this case, it is necessary to add some logic using setDataMask when you click on a region

**Comment by Yisel-Hernandez** [comment]
Buenas tardes a todos.

Lo que pasa es que estoy tratando de filtrar un gráfico de multiples capas, la idea que tuve fue en el mismo dashboard agregar los gráficos que componen el gráfico de múltiples capas, pero solo se filtran los gráficos por aparte pero no afecta en nada el de múltiples capas.

Alguno tiene idea de cómo podría hacer para que a medida de que se filtren los gráficos que componen el gráfico de múltiples capas este también se me filtre con los cambios de los demás gráficos?

Muchas gracias.

**Reply by romaiy** [reply]
Does this problem only apply to the country map chart or to some others?

**Reply by Yisel-Hernandez** [reply]
No, con los mapas de país no, es específicamente en el mapa Deck.gl Múltiples Capas, en el que se pueden unir varías capas

**Comment by gerenciaGif** [comment]
Having a map that automatically adjusts all other charts when you click on one state would be a significant improvement over color maps. In fact, what I do is create a second chart (the chart bar at the right in the image below) so by clicking on one state at chart triggers the adjustment in all other charts.
Thanks all!

<img width="985" height="331" alt="Captura desde 2025-12-03 10-51-45" src="https://github.com/user-attachments/assets/031cb9c6-c665-45af-b52f-09a1be6a4992" />
